src/form/field/form-field.test.tsx

```diff
@@ -0,0 +1,119 @@
+import React from "react";
+import {render, cleanup} from "@testing-library/react";
+import "@testing-library/jest-dom";
+import {create} from "react-test-renderer";
+
+import {testA11y} from "../../core/utils/test/testUtils";
+import FormField, {FormFieldProps} from "./FormField";
+import Input from "../input/Input";
+
+describe("<FormField />", () => {
+  afterEach(cleanup);
+
+  const defaultFormFieldProps: FormFieldProps = {
+    testid: "form-field",
+    children: <Input onChange={jest.fn} name={"test"} testid={"form-field.input"} />
+  };
+
+  it("should render correctly", () => {
+    render(<FormField {...defaultFormFieldProps} />);
+  });
+
+  it("should matches snapshot", () => {
+    const tree = create(<FormField {...defaultFormFieldProps} />).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("should pass a11y test", async () => {
+    const {container} = render(<FormField {...defaultFormFieldProps} />);
+
+    await testA11y(container, {rules: {label: {enabled: false}}});
+  });
+
+  it("should render children correctly", () => {
+    const {getByTestId} = render(<FormField {...defaultFormFieldProps} />);
+
+    const childrenContent = getByTestId("form-field.input");
+
+    expect(getByTestId("form-field")).toContainElement(childrenContent);
+  });
+
+  it("should render label correctly", () => {
+    const {getByTestId} = render(
+      <FormField
+        label={"Test"}
+        labelledBy={"form-field.labelled-by"}
+        labelFor={"form-field.input"}
+        {...defaultFormFieldProps}
+      />
+    );
+
+    const formFieldLabel = document.getElementsByTagName("label")[0];
+
+    expect(getByTestId("form-field")).toHaveTextContent("Test");
+    expect(formFieldLabel).toHaveAttribute("id", "form-field.labelled-by");
+    expect(formFieldLabel).toHaveAttribute("for", "form-field.input");
+  });
+
+  it("errorMessages prop should work correctly", () => {
+    const formFieldErrorMessages = ["Test Error 1", "Test Error 2"];

```

Comment:
What's the behavior if I provide `[]` or `[""]` as errorMessages ? Same question for the helperMessages. Should we test these too ?

Reply:
@edizcelik  What is the expected behavior here? Right now, if i provide `empty array []` it does not render `error-message-list` but if i provide `[""]` it render an empty `FormFieldMessage` . Is this the expected behavior or do we need to make changes here?